osu.Game/Overlays/Profile/Sections/DrawableHistoryItem.cs

```diff
@@ -0,0 +1,150 @@
+
+        [Resolved]
+        protected OverlayColourProvider ColourProvider { get; private set; }
+
+        protected readonly T Item;
+        private readonly (string name, Action action)[] properties;
+
+        private LinkFlowContainer linkFlow;
+
+        protected DrawableHistoryItem(T item)
+        {
+            Item = item;
+            properties = GetProperties();
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+            AddInternal(new GridContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                ColumnDimensions = new[]
+                {
+                    new Dimension(GridSizeMode.Absolute, size: LeftContentSize()),
+                    new Dimension(),
+                    new Dimension(GridSizeMode.AutoSize)
+                },
+                RowDimensions = new[]
+                {
+                    new Dimension(GridSizeMode.AutoSize)
+                },
+                Content = new[]
+                {
+                    new[]
+                    {
+                        CreateLeftContent(),
+                        linkFlow = new LinkFlowContainer(t => t.Font = OsuFont.GetFont(size: font_size))
+                        {
+                            Anchor = Anchor.CentreLeft,
+                            Origin = Anchor.CentreLeft,
+                            AutoSizeAxes = Axes.Y,
+                            RelativeSizeAxes = Axes.X,
+                        },
+                        new DrawableDate(GetDate())
+                        {
+                            Anchor = Anchor.CentreRight,
+                            Origin = Anchor.CentreRight,
+                            Colour = ColourProvider.Foreground1,
+                            Font = OsuFont.GetFont(size: font_size),
+                        }
+                    }
+                }
+            });
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            handleString(GetString());
+        }
+
+        protected abstract (string name, Action action)[] GetProperties();
+
+        protected abstract DateTimeOffset GetDate();
+
+        protected abstract string GetString();
+
+        protected virtual float LeftContentSize() => 0;
+
+        protected virtual Drawable CreateLeftContent() => Empty();
+
+        private void handleString(string s)
+        {
+            if (string.IsNullOrEmpty(s))
+                return;
+
+            if (!s.Contains(':'))
+            {
+                AddText(s);
+                return;
+            }
+
+            var variableIndex = s.IndexOf(':');
+
+            AddText(s.Substring(0, variableIndex));
+
+            var trimmed = handleVariable(s.Substring(variableIndex + 1));
+
+            handleString(trimmed);
+        }
```

Comment:
I don't think what the suggestion meant to convey is to literally copy the [localisation format from laravel](https://laravel.com/docs/7.x/localization)...

First off, *even if* this is the format that we want to be using, this code should not live here, it should be in a more general place. Secondly, I'd argue we should aim for a format which is more idiomatic to C# (something akin to `FormattableString`, but for drawables). Unfortunately I think rolling our own parsing would be unavoidable either way since `string.Format` seems inextensible to other types.

Seeing that the above needs a more general (maybe even framework-side) mechanism, I'd be for holding this change off entirely until that is dealt with.